frappe/public/js/frappe/form/toolbar.js

```diff
@@ -64,27 +68,81 @@ frappe.ui.form.Toolbar = Class.extend({
 	can_rename: function() {
 		return this.frm.perm[0].write && this.frm.meta.allow_rename && !this.frm.doc.__islocal;
 	},
-	setup_editable_title: function() {
-		var me = this;
-		this.page.$title_area.find(".title-text").on("click", function() {
-			if(me.is_title_editable()) {
-				frappe.prompt({fieldname: "title", fieldtype:"Data",
-					label: __("Title"), reqd: 1, "default": me.frm.doc.title },
-					function(data) {
-						if(data.title) {
-							me.frm.set_value("title", data.title);
-							if(!me.frm.doc.__islocal) {
-								me.frm.save_or_update();
-							} else {
-								me.set_title();
-							}
-						}
-					}, __("Edit Title"), __("Update"));
+	setup_editable_title: function () {
+		let me = this;
+
+		this.page.$title_area.find(".title-text").on("click", () => {
+			let fields = [];
+			let title_field = me.frm.meta.title_field;
+
+			// check if title is updateable
+			if (me.is_title_editable()) {
+				fields.push({
+					label: __("New {0}", [__(title_field)]),
+					fieldname: "title",
+					fieldtype: "Data",
+					reqd: 1,
+					default: me.frm.doc.title
```

Comment:
Does this work with DocTypes with `title_field` other than `title`?

Reply:
error on my part
thanks for picking it out!